R/BayesPrism.R

```diff
@@ -0,0 +1,125 @@
+#' @param n_cores Number of CPU threads used for parallel computing. Default=1
+#' @param n_cores_2g Number of CPU threads used for parallel computing for the final Gibbs sampling.
+#'  Default=NULL (same as ncores). Recommended to set to a number smaller than n.cores, if deconvolving
+#'  large number of mixtures, such as Visium data, or the number of cell types is large, to avoid memory overflow.
+#' @param first_gibbs_only A logical parameter denoting if to only run the first gibbs sampling,
+#' i.e. the initial estimates of theta and Z. Default: FALSE
+#' @param seed A numerical number specifying the random seed number to generate identical results
+#' between different runs. Default: NULL(ignore reproducibility).
+#'
+#' @return A list of results is returned including:
+#'   \item{para}{All input data and parameters.}
+#'   \item{res}{All output of TED. }
+#'   \item{res$first.gibbs.res$gibbs.theta}{Initial estimates of fraction for all cell subtypes in each bulk sample.}
+#'   \item{res$first.gibbs.res$Znkg}{Initial estimates of the mean of posterior read count for each cell subtypes  in each bulk sample.}
+#'   \item{res$first.gibbs.res$theta.merged}{Initial estimates of fraction summed across cell types in each bulk sample.}
+#'   \item{res$first.gibbs.res$Znkg.merged}{Initial estimates of the mean of posterior read count summed across cell types in each bulk sample.}
+#'   \item{res$Zkg.tum}{Mean of posterior of gene expression of tumor in each patient.}
+#'   \item{res$Zkg.tum.norm}{Depth normalized Zkg.tum (A pseudo count is added, such that the zero-valued genes have the same value as the min(phi.input)). Refered to as the psi.tum in the TED paper)}
+#'   \item{res$Zkg.tum.vst}{Variance stablized transformed value of Zkg.tum. If vst transformation is not feasible, return NULL. }
+#'   \item{res$phi.env}{Batch effect corrected expression profiles of stromal cells (refered to as the psi.str in the TED paper)}
+#'   \item{res$final.gibbs.theta}{Updated theta after batch correction and tumor expression estimates. This is the final deconvolution result.}
+#'   \item{res$cor.mat}{The correlation matrix of the estimated tumor expression profiles across bulk RNA-seq samples.}
+#' @export
+#'
+deconvolute_bayesprism <- function(bulk_gene_expression, single_cell_object, cell_type_annotations,
+                                   cell_subtype_labels = NULL, tum_key = NULL, pseudo_min = 1E-8, alpha = 1,
+                                   sigma = 2, outlier_cut = 0.01, outlier_fraction = 0.1,
+                                   gibbs_control = list(chain.length = 1000, burn.in = 500, thinning = 2),
+                                   opt_control = list(trace = 0, maxit = 100000), n_cores = 1,
+                                   n_cores_2g = NULL, first_gibbs_only = FALSE, seed = NULL) {
+
+  # package is required to run bayesprism, also with only a single core
+  require(snowfall)
```

Comment:
in a package one should never use `require` or `library`. 

instead use `@import` or `@importFrom` as a roxygen comment